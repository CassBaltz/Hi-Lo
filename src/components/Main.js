import React, { useState } from 'react';
import { withStyles } from '@material-ui/styles';
import { connect } from "react-redux";

import Game from './Game';
import Instructions from './Instructions';
import Scoreboard from "./Scoreboard";
import Winner from "./Winner";

import { startNewGame } from "../redux/actions/deck";
import getWinnerFromState from "../redux/util/getWinnerFromState";

export const Main = ({ dispatch, classes, deckId, winner, }) => {
  const [showInstructions, setShowInstructions] = useState(false);
  const newGame = () => {
    dispatch(startNewGame(deckId));
  };

  return (
    <React.Fragment>
      <div className={classes.header}>
        <button className={classes.instructions} onClick={() => setShowInstructions(true)}>INSTRUCTIONS</button>
        <button className={classes.newGame} onClick={newGame}>NEW GAME</button>
        <Scoreboard />
      </div>
      {showInstructions && <Instructions setShowInstructions={setShowInstructions} />}
      {winner && <Winner newGame={newGame} name={winner} />}
      <Game />
    </React.Fragment>
  );
};

const sharedButtonStyles = {
  boxShadow: "unset",
  border: "none",
  borderRadius: "3px",
  "&:focus": {
    outline: "none",
  },
  padding: "10px",
  cursor: "pointer",
};

const styles = {
  header: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    padding: "10px",
  },
  instructions: {
    ...sharedButtonStyles,
    backgroundColor: "#ffe8cc",
  },
  newGame: {
    ...sharedButtonStyles,
    backgroundColor: "#e3fafc",
  },
};

const mapStateToProps = state => ({
  deckId: state.deck.deckId,
  winner: !state.deck.resettingDeck ? getWinnerFromState(state) : null,
});

export default connect(mapStateToProps)(withStyles(styles)(Main));