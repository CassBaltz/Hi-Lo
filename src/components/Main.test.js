import React from "react";

import { Main } from "./Main";

const props = {
  classes: {
    header: "header",
    instructions: "instructions",
    newGame: "newGame",
  },
  dispatch: jest.fn(),
  deckId: "abc",
};

jest.mock("../redux/actions/deck", () => {
  return {
    __esModule: true,
    startNewGame: jest.fn(() => "abc"),
  };
});


describe("<Main />", () => {
  it("should display the main page with action buttons, a scoreboard, and the game", () => {
    const main = shallow(<Main {...props} />);
    expect(toJson(main)).toMatchSnapshot();
  });

  it("should call its dispatch function with the correct payload when the 'NEW GAME' button is clicked.", () => {
    const main = shallow(<Main {...props} />);
    const newGameButton = main.find("button.newGame");
    newGameButton.simulate("click");
    expect(props.dispatch).toHaveBeenCalledWith(props.deckId);
  });
});