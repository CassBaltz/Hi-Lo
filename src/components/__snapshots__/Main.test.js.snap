// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should display the main page with action buttons, a scoreboard, and the game 1`] = `
<Fragment>
  <div
    className="header"
  >
    <button
      className="instructions"
      onClick={[Function]}
    >
      INSTRUCTIONS
    </button>
    <button
      className="newGame"
      onClick={[Function]}
    >
      NEW GAME
    </button>
    <Connect(WithStyles(Scoreboard)) />
  </div>
  <Game />
</Fragment>
`;
